src/main/typescript/data/item/weapon/attack/source.ts

```diff
@@ -1,180 +1,53 @@
-import type { JSONSchemaType } from "ajv";
-import { SplashSize, SplashSizes } from "../../../../constants.js";
-import {
-  CompositeNumberSource,
-  COMPOSITE_NUMBER_SOURCE_JSON_SCHEMA
-} from "../../../common.js";
-import { TAGS_SOURCE_JSON_SCHEMA } from "../../common/baseItem/source.js";
+import { z } from "zod";
+import { zodToJsonSchema } from "zod-to-json-schema";
+import { SplashSizes } from "../../../../constants.js";
+import { COMP_NUM_SCHEMA } from "../../../common.js";
 
-export default class AttacksSource {
-  /** The source objects for the Attacks */
-  sources: Record<string, AttackSource> = {};
-}
+export type AttacksSource = z.infer<typeof ATTACKS_SCHEMA>;
+export type AttackSource = z.infer<typeof ATTACK_SCHEMA>;
+export type DamageSource = z.infer<typeof DAMAGE_SCHEMA>;
 
-export class AttackSource {
+export type DamageFallOffType = typeof DamageFallOffTypes[number];
+const DamageFallOffTypes = ["shotgun"] as const;
+
+/** The source data for weapon damage */
+export const DAMAGE_SCHEMA = z.object({
+  /** The base damage amount */
+  base: COMP_NUM_SCHEMA.default({}),
+
+  /** The number of d6 to throw for variable damage */
+  dice: COMP_NUM_SCHEMA.default({}),
```

Comment:
Those defaults don't seem correct. Might be worth to add a built-in default to the Composite Number schema directly.